mtp_noms_ops/apps/security/forms/check.py

```diff
@@ -74,17 +75,38 @@ def get_object_list(self):
         return object_list
 
 
-class ActionCheckForm(GARequestErrorReportingMixin, forms.Form):
+class AcceptOrRejectCheckForm(GARequestErrorReportingMixin, forms.Form):
     """
     Base CheckForm for accepting or rejecting a check.
     """
-    non_pending_error_msg = NotImplementedError
+    URGENT_IF_OLDER_THAN = datetime.timedelta(days=3)
+    decision_reason = forms.CharField(
+        label=gettext_lazy('Give details (details are optional when accepting)'),
+        required=False,
+    )
+    fiu_action = forms.CharField(max_length=10)
+
+    non_pending_error_msg = "You cannot action this credit as it's not in pending"
 
     def __init__(self, object_id, request, **kwargs):
         super().__init__(**kwargs)
         self.object_id = object_id
+        self.need_attention_date = self.get_need_attention_date()
         self.request = request
 
+    def get_need_attention_date(self):
```

Comment:
this is now a duplicate of the same method in `CheckListForm`, right? probably makes sense to extract it into a separate function that both forms can use. there'd be less duplication to manage in the future

Reply:
I started but didn't finish that actually. Will see how it goes now. Main problem I had was where to put it that was sensible.

Reply:
I've seen the pattern I used in other Rest Framework projects I looked at, but keen on your thoughts on whether I'm matching your or other contributors patterns for sticking reusable functions before classes that use them - or is there a better place for it and import it for use?